pkg/ingester/user_state.go

```diff
@@ -271,6 +275,10 @@ func (u *userState) forSeriesMatching(ctx context.Context, allMatchers []*labels
 
 	level.Debug(log).Log("series", len(fps))
 
+	iter := 0
+	if callback2Interval <= 0 {
+		callback2Interval = 1
+	}
 	// fps is sorted, lock them in order to prevent deadlocks
 outer:
 	for _, fp := range fps {
```

Comment:
Instead of `iter`, you could use `i` from this loop and do `i % batchSize == 0` for the test.